lib/SearchAndSort/SearchAndSort.js

```diff
@@ -398,16 +398,20 @@ class SearchAndSort extends React.Component {
       showSingleResult,
       finishedResourceName,
       stripes: { logger },
+      location,
     } = this.props;
 
     const previousState = makeConnectedSource(prevProps, logger);
     const currentState = makeConnectedSource(this.props, logger);
 
+    const recordId = location.pathname.split('/')[3];
+
     // if the results list is winnowed down to a single record, display the record.
     // If there's a single hit, open the detail pane as side-effect.
     const totalCountResult = currentState.totalCount();
     if (showSingleResult &&
       totalCountResult === 1 &&
+      !recordId &&
```

Comment:
Technically, this looks like the correct change here, but you must include some comments to explain what's going on and why.

Reply:
Yes, that will make it clearer, thanks @zburke!